### PR TITLE
Fix kubernetes dev workflows

### PR DESCRIPTION
**Pull request by yashgorana**
## Description

Closes https://github.com/OpenMined/Heartbeat/issues/913

Implemented as a part of
- https://github.com/OpenMined/Heartbeat/issues/892
- https://github.com/OpenMined/Heartbeat/issues/894

Fixes:
- Hot reload in Kubernetes
- Image not found errors in hot reloading
- Static credentials when deploying syft
- tox `dev.k8s.registry` command asking for sudo password

Adds:
- `tox -e dev.k8s.hotreload` to start hot reloading
- `tox -e dev.k8s.cleanup` to cleanup deployment while keeping the cluster up

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by yashgorana** [comment]
Devs are expected to run the commands in the following order:

For starting the cluster & registry (once first time or after destroying)
- `tox -e dev.k8s.start` to start a local registry + cluster and patch hosts file

For deploying syft (every time after updating code/helm charts)
- `tox -e dev.k8s.deploy` to deploy syft on the cluster
- `tox -e dev.k8s.hotreload` to deploy syft on the cluster with hot reloading and port forwards

For cleaning up
- `tox -e dev.k8s.cleanup` only removes the syft deployment, images, etc but keep the cluster running
- `tox -e dev.k8s.destroy` deletes the deployment, cluster, but keeps the registry running
- `tox -e dev.k8s.destroyall` deletes deployment, cluster and registry

File: packages/grid/devspace.yaml [hunk 1]
```diff
@@ -68,28 +65,34 @@ deployments:
...
           registry: ${CONTAINER_REGISTRY}
-          version: "dev-latest"
+          version: dev-${DEVSPACE_TIMESTAMP}
         node:
           settings:
             nodeName: ${NODE_NAME}
             nodeType: "domain"
-        # configuration:
-        #   devmode: True
+        configuration:
```
Comment (rasswanth-s) [inline packages/grid/devspace.yaml:73]: We need the dev mode to be enabled during only dev.k8s.hotreload, and  disable during dev.k8s.deploy

Comment (yashgorana) [reply packages/grid/devspace.yaml:73]: see the comment on this issue https://github.com/OpenMined/Heartbeat/issues/913

**Review by rasswanth-s — APPROVED** [review]
Well Done @yashgorana

**Pull request merged**